lib/redis_cluster/pool.rb

```diff
@@ -54,8 +54,30 @@ def pipelined(args, &block)
       random_node.execute :pipelined, args, &block
     end
 
+    # Implements scan across all nodes in the pool.  Encodes the current node in the returned cursor.
+    # Cursors will behave strangely if the node list changes during iteration.
+    def scan(args)
+      cursor = args.first
+      options = args[1] || {}
+      orig_cursor, node_index = decode_scan_cursor(cursor)
+      next_cursor, result = @nodes[node_index].execute("scan", [orig_cursor.to_s, options])
+      [ encode_scan_cursor(next_cursor.to_i, node_index), result ]
+    end
+
     private
 
+    def encode_scan_cursor(scan_cursor, targeted_node)
+      if scan_cursor == 0
+        ((targeted_node + 1) % @nodes.size).to_s
+      else
+        ((scan_cursor * @nodes.size) + targeted_node).to_s
+      end
```

Comment:
Odd that cursors are sometimes int and sometimes strings. The `.to_s` would be DRY after the `end` here. Or maybe even better would be to put the `.to_s` in the calling code? Ideally that calling could would handle the translation to/from string and all the rest of the code would just work with integers.

Reply:
Woo!  Thanks for the code review....

I agree.  Redis promises that they are integers, but returns and requires string arguments.   I renamed some things to make them clearer.  A scan_cursor is the public cursor that encodes a node_cursor and a node_index.  I also change the decode method to be more consistent.  Node cursors and scan cursors are always returned as strings.